module/zfs/zvol.c

```diff
@@ -1019,6 +1018,19 @@ zvol_shutdown_zv(zvol_state_t *zv)
 }
 
 /*
+ * return the proper tag for rollback and recv
+ */
+void *
+zvol_tag(zvol_state_t *zv)
+{
+	ASSERT(RW_WRITE_HELD(&zv->zv_suspend_lock));
+	if (zv->zv_open_count > 0)
+		return (zv);
+	else
+		return (NULL);
```

Comment:
[style] Personally I wouldn't object to reworking this so it's a little more concise.  But I'm OK with either way, pick your favorite.  I agree this helper function is a nice way to handle this and it let's us verify the zv_suspend_lock is held.

```
return ((zv->zv_open_count > 0) ? zv : NULL);
```